src/helpers.rs

```diff
@@ -509,3 +509,39 @@ fn bytes_to_os_str<'tcx, 'a>(bytes: &'a [u8]) -> InterpResult<'tcx, &'a OsStr> {
         .map_err(|_| err_unsup_format!("{:?} is not a valid utf-8 string", bytes))?;
     Ok(&OsStr::new(s))
 }
+
+pub fn int_to_immty_checked<'tcx>(
```

Comment:
Following `ImmTy::from_int`, IMO a better name would be `immty_from_int_checked`. (Same for `uint`).